channels/applicationinsights-channel-js/src/Statsbeat.ts

```diff
@@ -0,0 +1,203 @@
+import { NetworkStatsbeat } from "./NetworkStatsbeat";
+import { Sender } from "./Sender";
+import { StatsbeatCounter } from "./Constants";
+import { IConfig, Metric } from "@microsoft/applicationinsights-common";
+import { IConfiguration, IAppInsightsCore, IPlugin, ITelemetryPluginChain, ITelemetryItem, objKeys, dateNow,  } from '@microsoft/applicationinsights-core-js';
+import dynamicProto from '@microsoft/dynamicproto-js';
+
+export class Statsbeat {
+    public static INSTRUMENTATION_KEY = "c4a29126-a7cb-47e5-b348-11414998b11e";
+    public static STATS_COLLECTION_SHORT_INTERVAL: number = 900000; // 15 minutes
+    public static NETWORK = "Network";
+
+    constructor() {
+        let _networkCounter: NetworkStatsbeat;
+        let _sender: Sender;
+        let _handle: any;
+        let _statsbeatMetrics: { properties?: {} };
+        let _config: IConfiguration & IConfig;
+        let _isEnabled: boolean;
+
+        // Custom dimensions
+        let _cikey: string;
+        dynamicProto(Statsbeat, this, (_self, _base) => {
+            _self.initialize = (config: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?:ITelemetryPluginChain, endpoint?: string) => {
+                _networkCounter = new NetworkStatsbeat(endpoint);
+                _sender = new Sender(undefined);
+                let senderConfig = {...config};
+                senderConfig.instrumentationKey = Statsbeat.INSTRUMENTATION_KEY;
+                _sender.initialize(senderConfig, core, extensions, pluginChain, endpoint);
+                _statsbeatMetrics = {};
+                _config = config;
+                _isEnabled = _config.disableStatsbeat !== true;
```

Comment:
Is there a point to all the initializing code above this line if `disableStatsbeat` is `true`?

Reply:
I realized when it comes into statsbeat's initialize function, the config.disableStatsbeat is either `false` or `undefined`, it will never be `true`. Because in sender, we only call statsbeat.initialize when the config is false or undefined. But the flag is useful to check if statsbeat is enabled. So to answer Leighton's question - the sender needs to be initialized for statsbeat because this is inside statsbeat's initialize function, but as it confuses you, it may not necessary to check config, just assign `_isEnabled = true` here would be good I think. @MSNev what do you think?

Reply:
Yeh, just assign to true.